extensions/amp-analytics/0.1/test/test-analytics-root.js

```diff
@@ -333,25 +333,106 @@ describes.realWin('AmpdocAnalyticsRoot', {amp: 1}, (env) => {
       addTestInstance(root3.getElement(body, '#target'), null);
     });
 
-    it('should find an AMP element for AMP search', () => {
+    it('should find an AMP element for AMP search', async () => {
       child.classList.add('i-amphtml-element');
-      return root.getAmpElement(body, '#child').then((element) => {
-        expect(element).to.equal(child);
-      });
+      const element = await root.getAmpElement(body, '#child');
+      expect(element).to.equal(child);
     });
 
-    it('should allow not-found element for AMP search', () => {
-      return root.getAmpElement(body, '#unknown').catch((error) => {
+    it('should handle missing selector for AMP search', async () => {
+      await root.getAmpElement(body, '#unknown').catch((error) => {
         expect(error).to.match(/Element "#unknown" not found/);
       });
     });
 
-    it('should fail if the found element is not AMP for AMP search', () => {
+    it('should fail if the found element is not AMP for AMP search', async () => {
       child.classList.remove('i-amphtml-element');
-      return root.getAmpElement(body, '#child').catch((error) => {
+      await root.getAmpElement(body, '#child').catch((error) => {
         expect(error).to.match(/required to be an AMP element/);
       });
     });
+
+    describe('get amp elements', () => {
+      let child2;
+      let child3;
+
+      beforeEach(() => {
+        child2 = win.document.createElement('child');
+        child3 = win.document.createElement('child');
+        body.appendChild(child2);
+        body.appendChild(child3);
+        child.classList.add('i-amphtml-element');
+        child2.classList.add('i-amphtml-element');
+        child3.classList.add('i-amphtml-element');
+      });
+
+      it('should find element and elements by selector', async () => {
+        child.classList.add('myClass');
+        child2.classList.add('myClass');
+        child3.classList.add('notMyClass');
+        expect(
+          await root.getAmpElements(body, ['.myClass'], null, true)
+        ).to.deep.equal([child, child2]);
+        expect(
+          await root.getAmpElements(body, '.notMyClass', null, false)
+        ).to.deep.equal([child3]);
+      });
+
+      it('should remove duplicate elements found', async () => {
+        child.id = 'myId';
+        child.classList.add('myClass');
+        expect(
+          await root.getAmpElements(body, ['.myClass', '#myId'], null, true)
+        ).to.deep.equal([child]);
+      });
+
+      it('should remove duplicate selectors', async () => {
+        child.classList.add('myClass');
+        expect(
+          await root.getAmpElements(body, ['.myClass', '.myClass'], null, true)
```

Comment:
The check for duplicate selectors is not within `rootAmpElements`, you may want to move the test to test-event.js?

Reply:
Good call.